### PR TITLE
Add Timestamp Data Types

### PR DESCRIPTION
**Pull request by ajschmidt8**
This PR adds the following Series data types:

- `Series.TimestampDay`
- `Series.TimestampSecond`
- `Series.TimestampMillisecond`
- `Series.TimestampNanosecond`
- `Series.TimestampMicrosecond`

### CODE REVIEW

File: modules/cudf/test/cudf-series-test.ts [hunk 1]
```diff
@@ -469,3 +480,68 @@ ${[true, null, true]} | ${[true, true, true]}
 
   expect([...result]).toEqual(expected);
 });
+
+test('Series.TimestampDay', () => {
+  const dateTime = Math.floor(new Date(2021, 4, 13, 20, 11, 30, 100).getTime() / 86400000);
+  const s        = Series.new({type: new TimestampDay, data: new Int32Buffer([dateTime])});
```
Comment (ajschmidt8) [inline modules/cudf/test/cudf-series-test.ts:486]: not sure how to make these tests work without specifying `new Int32Buffer` in the `data` key here :thinking:

Comment (trxcllnt) [reply modules/cudf/test/cudf-series-test.ts:486]: 1. Add another [`Series.new()` overload](https://github.com/rapidsai/node-rapids/blob/781d9f12465ce0d570e14e91cf5f67794b430f3b/modules/cudf/src/series.ts#L231-L242) for `input: (Date|null|undefined)[]): Series<TimestampMillisecond>`
  a. Don't forget to add one for lists of Dates! ([example](https://github.com/rapidsai/node-rapids/blob/781d9f12465ce0d570e14e91cf5f67794b430f3b/modules/cudf/src/series.ts#L285-L298))
2. Check for JS Date in the [`inferType()` function](https://github.com/rapidsai/node-rapids/blob/781d9f12465ce0d570e14e91cf5f67794b430f3b/modules/cudf/src/series.ts#L916-L983) and return a `TimestampMillisecond` instance.

File: modules/cudf/src/column/from_arrow.ts [hunk 2]
```diff
@@ -134,12 +133,35 @@ class VectorToColumnVisitor extends arrow.Visitor {
...
+  }
+  visitDateMillisecond<T extends arrow.DateMillisecond>(
+    {length, data: {values: data, nullBitmap: nullMask}}: arrow.Vector<T>) {
+    return new Column(
+      {type: new TimestampMillisecond, length, data: data.subarray(0, length), nullMask});
+  }
+  visitTimestampSecond<T extends arrow.TimestampSecond>(
+    {length, data: {values: data, nullBitmap: nullMask}}: arrow.Vector<T>) {
+    return new Column(
+      {type: new TimestampSecond, length, data: data.subarray(0, length), nullMask});
```
Comment (trxcllnt) [inline modules/cudf/src/column/from_arrow.ts:148]:
```suggestion
      {type: new TimestampSecond, length, data: data.subarray(0, length * 2), nullMask});
```

File: modules/cudf/src/column/from_arrow.ts [hunk 3]
```diff
@@ -134,12 +133,35 @@ class VectorToColumnVisitor extends arrow.Visitor {
...
+  }
+  visitTimestampMillisecond<T extends arrow.TimestampMillisecond>(
+    {length, data: {values: data, nullBitmap: nullMask}}: arrow.Vector<T>) {
+    return new Column(
+      {type: new TimestampMillisecond, length, data: data.subarray(0, length), nullMask});
```
Comment (trxcllnt) [inline modules/cudf/src/column/from_arrow.ts:153]:
```suggestion
      {type: new TimestampMillisecond, length, data: data.subarray(0, length * 2), nullMask});
```

File: modules/cudf/src/column/from_arrow.ts [hunk 4]
```diff
@@ -134,12 +133,35 @@ class VectorToColumnVisitor extends arrow.Visitor {
...
+  }
+  visitTimestampMicrosecond<T extends arrow.TimestampMicrosecond>(
+    {length, data: {values: data, nullBitmap: nullMask}}: arrow.Vector<T>) {
+    return new Column(
+      {type: new TimestampMicrosecond, length, data: data.subarray(0, length), nullMask});
```
Comment (trxcllnt) [inline modules/cudf/src/column/from_arrow.ts:158]:
```suggestion
      {type: new TimestampMicrosecond, length, data: data.subarray(0, length * 2), nullMask});
```

File: modules/cudf/src/column/from_arrow.ts [hunk 5]
```diff
@@ -134,12 +133,35 @@ class VectorToColumnVisitor extends arrow.Visitor {
...
+  }
+  visitTimestampNanosecond<T extends arrow.TimestampNanosecond>(
+    {length, data: {values: data, nullBitmap: nullMask}}: arrow.Vector<T>) {
+    return new Column(
+      {type: new TimestampNanosecond, length, data: data.subarray(0, length), nullMask});
```
Comment (trxcllnt) [inline modules/cudf/src/column/from_arrow.ts:163]:
```suggestion
      {type: new TimestampNanosecond, length, data: data.subarray(0, length * 2), nullMask});
```

File: modules/cudf/src/series/timestamp.ts [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+   */
+  getValue(index: number) {
+    const val = this._col.getValue(index);
+    return val === null ? null : new Date(val * 86400000);
+  }
+}
+
+export class TimestampSecondSeries extends TimestampSeries<TimestampSecond> {
+  get data() {
+    return new Int32Buffer(this._col.data).subarray(this.offset, this.offset + this.length);
```
Comment (trxcllnt) [inline modules/cudf/src/series/timestamp.ts:78]:
```suggestion
    return new Int64Buffer(this._col.data).subarray(this.offset, this.offset + this.length);
```

**Comment by mzegar** [comment]
LGTM on my end, nothing to comment. Going to leave it to others to approve.

File: modules/cudf/src/series.ts [hunk 7]
```diff
@@ -922,45 +961,56 @@ function inferType(value: any[]): DataType {
...
-      case 'object': return Array.isArray(val) ? ++arraysCount : ++objectsCount;
+      case 'object':
+        if (Array.isArray(val)) {
+          ++arraysCount;
+        } else if (Object.prototype.toString.call(val) === '[object Date]') {
+          ++datesCount;
+        } else {
+          ++objectsCount;
+        }
+        return;
```
Comment (trxcllnt) [inline modules/cudf/src/series.ts:980]: This should fall through if not one of our recognized types, right?

```suggestion
      case 'object':
        if (Array.isArray(val)) {
          return ++arraysCount;
        } else if (Object.prototype.toString.call(val) === '[object Date]') {
          return ++datesCount;
        } else {
          return ++objectsCount;
        }
```

**Review by trxcllnt — CHANGES_REQUESTED** [review]
One small suggestion in the `inferType()` function, other than that LGTM

**Comment by ajschmidt8** [comment]
> One small suggestion in the `inferType()` function, other than that LGTM

committed!

**Review by trxcllnt — APPROVED** [review]

**Pull request merged**